### PR TITLE
Tests for Const (mostly copypasta from IListTests)

### PR DESCRIPTION
**Pull request by diminou**
Related to #1959 
As the title says, it is mostly copypasta from IListTests, and thus may be lacking in checking crucial Const-specific properties. Please point this out if it is the case.

### CODE REVIEW

**Comment by edmundnoble** [comment]
I recommend adding a test for `Phantom` laws (`FunctorLaws.Phantom.identityToIdentity`) where the assertion is `_ eq _`. Also, I'd suggest replacing some of the `assertEqual(x, y)` calls with `assert(x eq y)`, just because we care that `Const.run` and `Const.apply` actually don't allocate.

**Comment by edmundnoble** [comment]
I took the liberty of fixing a merge conflict.

File: tests/src/main/scala/scalaz/tests/ConstTests.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+final class ConstTests {
+
+  val consts = List(
+    Const[List[Int], List[Double]](List(1)),
+    Const[List[Int], List[Double]](List())
+  )
+
+  def cross[A, B](l1: List[A], l2: List[B]): List[(A, B)] =
+    l1.flatMap(a1 => l2.map(a2 => (a1, a2)))
```
[the remark this answers is not in the document]
Comment (emilypi) [reply tests/src/main/scala/scalaz/tests/ConstTests.scala]: I'm good with that :)

**Comment by emilypi** [comment]
Given all of our thumbs, I'm merging

**Comment by diminou** [comment]
Oh wow, thanks @edmundnoble @emilypi and @alexknvl for your work on this PR.
I'd like to add tests for Phantom laws that @edmundnoble suggested, but, having little time to myself ATM, I believe it would be better to create a separate tiny PR for this later.

**Comment by alexknvl** [comment]
Hmm, can you merge the most recent master (series/8.0.x) into the PR? The diff looks weird, I don't get why.

**Comment by diminou** [comment]
@alexknvl perhaps I should have merged and not rebased. I could rewind to 0515c41 and then merge.

**Comment by alexknvl** [comment]
If that's not too much work.

**Pull request merged**